src/client/discord.h

```diff
@@ -0,0 +1,57 @@
+/*
+Minetest
+Copyright (C) 2018 Dumbeldor, Vincent Glize <vincent.glize@live.fr>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#pragma once
+#include <iostream>
+#include <memory>
+
+struct DataRichPresence
```

Comment:
This struct is and will only used in `discord.cpp` - move the definition there to reduce the header size.

Reply:
Yes, thanks !

Reply:
No it is not possible. Unless you turn m_data into a pointer.
You can't declare a member of an incomplete type, because the compiler doesn't know how big it is and how much space should it reserve for it (among other things).

Reply:
use a unique_ptr then ?

Reply:
we can do like that then. Even if i find it a shame to manage a pointer unnecessarily